src/hal/i_components/knob2float.icomp

```diff
@@ -34,8 +34,8 @@ license "GPL";
 
 FUNCTION(_)
 {
-    long delta_counts;
-    double tmp_out;
+    hal_float_t delta_counts;
```

Comment:
should be hal_s32_t, line 45 https://github.com/machinekit/machinekit/pull/694/files#diff-0c5675002ab9f467e09dcb64e1772b6aR45 contains an implicit type conversion that should be marked with a typecast